cpp/tests/buffer.cpp

```diff
@@ -4,6 +4,7 @@
  */
 #include <algorithm>
 #include <numeric>
+#include <tuple>
 #include <utility>
```

Comment:
```suggestion
#include <utility>
```

`std::ignore` should already be in `<utility>` (since C++11), so no need for the `<tuple>` include I think?